### DISCUSSION TITLE
Flet for Django

### DISCUSSION THREAD

**Discussion by jet10000** [flet-dev/flet#1778] [General]
https://flet.dev/blog/flet-for-fastapi

Nice to see flet's progress. 🍻🍻

Looking forward to integrating with django.

**Comment by FeodorFitsner** [comment]
What would be the benefit of that integration?

**Reply by baseplate-admin** [reply]
Hi, let me answer this

> benefit of that integration?

For starters, django is used by large corporations and has a relatively large community compared to `fastapi`.

It has a well established ecosystem of diverse packages but lacks a good SPA framework. I believe that `flet` can fill that gap. 

Going back you already implemented most of the logics. If you can port it to `asgiref`, ( therefore targeting the [`ASGI` protocol](https://en.wikipedia.org/wiki/Asynchronous_Server_Gateway_Interface) ) flet can become framework independent :)

**Reply by FeodorFitsner** [reply]
Sorry for the confusion, I probably meant how would you use Flet with Django (or Django with Flet), like what's a use case?

**Reply by baseplate-admin** [reply]
> Sorry for the confusion

I am sorry for not being clear here. 

> how would you use Flet with Django

Quoting your statement on the blog, "On the other hand, seasoned FastAPI developers can use Flet to easily add interactive, real-time dashboards and admin UI to their existing or new FastAPI services."

We can use `django` to build interactive, real-time dashboards to existing `django` services.


`django` is another web framework ( that's used in websites like `Instagram`, `mozilla`, `youtube` and `google` ) that's very popular

**Reply by FeodorFitsner** [reply]
Sounds right, thanks for elaborating on this!

**Reply by FeodorFitsner** [reply]
I've added a new issue: https://github.com/flet-dev/flet/issues/1782

**Reply by baseplate-admin** [reply]
Thank you very much..


I have one more request.

You should look into making flet into an asgi application. That way you dont have to write logics for each web framework :D

**Comment by jet10000** [comment]
https://github.com/pauloxnet/uDjango

**Comment by jet10000** [comment]
up

**Reply by abcnow** [reply]
@FeodorFitsner .. I think fastapi is pretty good for small apps but you have to set up everything and if you are doing login/registration you have to create that from scratch..
Django have already those things integrated and it is easier to deal with databases and migration..

**Reply by baseplate-admin** [reply]
I also think it's preferable to code in a framework agnostic way :)

**Comment by jet10000** [comment]
https://django-ninja.dev/

**Comment by jet10000** [comment]
<img width="748" alt="image" src="https://github.com/flet-dev/flet/assets/2258120/d7fc4e6a-bc03-47b0-bc79-43540efb8441">
